ZenPacks/zenoss/MySqlMonitor/modeler/plugins/MySQLCollector.py

```diff
@@ -90,28 +91,77 @@ def collect(self, device, log):
 
             res = {}
             res["id"] = "{0}_{1}".format(el.get("user"), el.get("port"))
-            for key, query in self.queries.iteritems():
-                try:
-                    res[key] = yield dbpool.runQuery(query)
-                except Exception, e:
-                    self.is_clear_run = False
-                    res[key] = ()
-                    msg, severity = self._error(
-                        str(e), el.get("user"), el.get("port"))
-
-                    log.error(msg)
-
-                    if severity == 5:
-                        self._send_event(msg, device.id, severity)
-                        dbpool.close()
-                        defer.returnValue('Error')
-                        return
+
+            # Perform the simple SQL queries
+            yield self._doSimpleQueries(dbpool, res)
+            if self.is_clear_run is False:
+                dbpool.close()
+                defer.returnValue('Error')
+                return
+            # Perform the more complex or special SQL queries
+            yield self._doReplicaQuery(dbpool, res)
+            if self.is_clear_run is False:
+                dbpool.close()
+                defer.returnValue('Error')
+                return
 
             dbpool.close()
             result.append(res)
 
         defer.returnValue(result)
 
+    @defer.inlineCallbacks
+    def _doSimpleQueries(self, dbpool, res):
+        for key, query in self.queries.iteritems():
+            # non str type queries identify them as complex ones 
+            if not isinstance(query, str):
+                continue
+            try:
+                res[key] = yield dbpool.runQuery(query)
+            except Exception, e:
+                self.is_clear_run = False
+                res[key] = ()
+                msg, severity = self._error(
+                    str(e), el.get("user"), el.get("port"))
+                log.error(msg)
+                if severity == 5:
+                    self._send_event(msg, device.id, severity)
+
+    @defer.inlineCallbacks
+    def _doReplicaQuery(self, dbpool, res):
+        if 'replica' not in self.queries:
+            return
+
+        if 'version' not in res:
+            query = self.queries['replica'][0]
+        numVer = re.match(
+            '(\d+)\.(\d+)\.(\d+)-.*',
+            self._version(res['version'])
+        )
+        if not numVer:
+            self.is_clear_run = False
+            res['replica'] = ()
+            msg = "Problem parsing/comparing MySQL version"
+            severity = 4
+            log.error(msg)
+            self._send_event(msg, device.id, severity)
+        
+        numVer = tuple(int(x) for x in numVer.groups())
```

Comment:
if numVer is None (no re match), numVer.groups() will fail with attribute error.